rules/ast/functions.go

```diff
@@ -498,6 +499,43 @@ func derivImpl(timestamp clientmodel.Timestamp, args []Node) interface{} {
 	return resultVector
 }
 
+// === histogram_quantile(k ScalarNode, vector VectorNode) Vector ===
```

Comment:
>WRT by clauses in other cases: sum(m) works just fine without by clause. If we really want to include aggregation into histogram_quantile, it wolud be inconsistent if it would not work without by clause.

histogram_quantile should work without a by clause - in the same way `sum` does by applying to everything as one metric. If you applied the logic you're proposing to sum/min/max they'd all be the identity function.

topk/bottomk should take a by clause, I'm working under the assumption that we'll add one at some point.